python/servo/bootstrap.py

```diff
@@ -29,6 +51,9 @@ def salt(context, force=False):
     process.wait()
     if process.returncode:
         out, err = process.communicate()
+        if "pycrypto/setup.py" in err and not retry:
+            return install_salt_dependencies(context, force)
```

Comment:
Hmm, I don't know where my previous comment about this went.

Checking for a substring of the error message in this way seems fragile to me; Salt has changed crypto libraries in the past.

The mutual recursion here is error-prone as well; if in the future there is a new system-level dependency for Salt, then this code won't install it, causing the error on each Salt invocation and endlessly recursing through `salt` and `install_salt_dependencies` in a loop, causing the process to hang.

I still think it's better to check for the the Salt dependencies on each run before installing Salt. We can make this faster by using `dpkg -s <package> <names> <here>` first, and only running `apt-get install` if it returns nonzero; I'm not sure about the CentOS/Fedora equivalent.